packages/react-aria-components/docs/Virtualizer.mdx

```diff
@@ -0,0 +1,430 @@
+      <li><strong>Layout objects must be memoized.</strong> Layouts store cached data that must not be reset on every React render.</li>
+      <li><strong>Virtualized components must have a defined size.</strong> This may be an explicit CSS `width` and `height`, or an implicit size (e.g. percentage or `flex`) bounded by an ancestor element. Without a bounded size, all items will be rendered to the DOM, negating the performance benefits of virtualized scrolling.</li>
+    </ol>
+  </Content>
+</InlineAlert>
+
+## Layouts
+
+Virtualizer uses <TypeLink links={docs.links} type={docs.exports.Layout} /> objects to determine the position and size of each item, and provide the list of currently visible items. When using a Virtualizer, all items are positioned by the `Layout`, and CSS layout properties such as flexbox and grid do not apply.
+
+### ListLayout
+
+`ListLayout` supports layout of items in a vertical stack. Rows can be fixed or variable height. When using variable heights, set the `estimatedRowHeight` to a reasonable guess for how tall the rows will be on average. This allows the size of the scrollbar to be calculated. `ListLayout` supports the following options:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType {...docs.exports.ListLayoutOptions} />
+</TypeContext.Provider>
+
+This example shows a [GridList](GridList.html) with variable height rows due to text wrapping.
+
+```tsx import
+let lorem = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin sit amet tristique risus. In sit amet suscipit lorem. Orci varius natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. In condimentum imperdiet metus non condimentum. Duis eu velit et quam accumsan tempus at id velit. Duis elementum elementum purus, id tempus mauris posuere a. Nunc vestibulum sapien pellentesque lectus commodo ornare.'.split(' ');
+let items = [];
+for (let i = 0; i < 5000; i++) {
+  let words = Math.max(2, Math.floor(Math.random() * 25));
+  let name = lorem.slice(0, words).join(' ');
+  items.push({id: i, name});
+}
+```
+
+```tsx example
+import {Virtualizer, ListLayout} from 'react-aria-components';
+import {MyGridList, MyItem} from './GridList';
+
+function Example() {
+  /*- begin highlight -*/
+  let layout = React.useMemo(() => new ListLayout({
+    estimatedRowHeight: 34,
+    gap: 4,
+    padding: 4
+  }), []);
+  /*- end highlight -*/
+
+  return (
+    <Virtualizer layout={layout}>
+      <MyGridList aria-label="Virtualized GridList" selectionMode="multiple" items={items}>
+        {item => <MyItem>{item.name}</MyItem>}
+      </MyGridList>
+    </Virtualizer>
+  );
+}
+```
+
+### GridLayout
+
+`GridLayout` supports layout of items in an equal size grid. The items are sized between a minimum and maximum size depending on the width of the container. It supports the following options:
+
+<TypeContext.Provider value={docs.links}>
+  <InterfaceType {...docs.exports.GridLayoutOptions} />
```

Comment:
for some reason, some sizes render as 200 x 200, but others are just Infinity, with no cross dimension

Reply:
where?

Reply:
https://reactspectrum.blob.core.windows.net/reactspectrum/4dc8819bc6405886e263049d98a1610aec8adf46/docs/react-aria/Virtualizer.html#gridlayout
![Screenshot 2025-02-05 at 12 19 28 pm](https://github.com/user-attachments/assets/7f43b855-1370-422b-88ef-d4ec1315035f)

Reply:
oh ehhh it's so long though. Is it not obvious enough?

Reply:
I'm fine with it, honestly, more weird that it's happening, I'm not sure why this would be, haha

Reply:
oh it just comes from the doc comment